### PR TITLE
[PAGC-1757] Re-introduce support for non-root routes and `aoi` tile clipping

### PR DESCRIPTION
**Pull request by DanSchoppe**
**Reminder: Public repo**

## What / Why

This PR re-introduces [the changes](https://github.com/developmentseed/titiler/compare/0.6.0...SenteraLLC:path-prefix) from #1 to support an `aoi` tile clipping argument, and #2 to support hosting TiTiler on a non-root route. This work is being done in tandem with SenteraLLC/rio-tiler#2 to modernize the SenteraLLC fork of TiTiler.

I also wrote a bit of content in a new wiki: https://github.com/SenteraLLC/titiler/wiki

## QA Strategy
- [ ] Merge Latest Master

### CODE REVIEW

File: src/titiler/core/pyproject.toml [hunk 1]
```diff
@@ -35,7 +35,7 @@ dependencies = [
     "numpy",
     "pydantic",
     "rasterio",
-    "rio-tiler>=4.1.6,<4.2",
+    "rio-tiler@https://github.com/SenteraLLC/rio-tiler/archive/refs/heads/4.1.10-upgrade-dev.tar.gz",
```
Comment (DanSchoppe) [inline src/titiler/core/pyproject.toml]: 🚨 Update after SenteraLLC/rio-tiler#2 merges and `4.1.10-0` release is created 🚨

```suggestion
    "rio-tiler@https://github.com/SenteraLLC/rio-tiler/archive/refs/tags/4.1.10-0.tar.gz",
```

File: pyproject.toml [hunk 2]
```diff
@@ -29,10 +29,12 @@ classifiers = [
 ]
 version="0.11.6"
 dependencies = [
-    "titiler.core==0.11.6",
-    "titiler.extensions==0.11.6",
-    "titiler.mosaic==0.11.6",
-    "titiler.application==0.11.6",
+    # Configured for volume-mount path of Lambda Dockerfile. For local
+    # development, update accordingly for your local filesystem path:
+    "titiler.core @ file:///titiler/src/titiler/core",
```
Comment (jdalt) [inline pyproject.toml:34]: How did this work before? Why did you change to the `@ file:///` style?

File: src/titiler/application/titiler/application/main.py [hunk 3]
```diff
@@ -40,14 +41,34 @@
...
 
+LEVELS = {
+    "debug": logging.DEBUG,
+    "info": logging.INFO,
+    "warning": logging.WARNING,
+    "error": logging.ERROR,
+    "critical": logging.CRITICAL,
+}
+# Remove any AWS-injected logger handlers to fix Lambda logging to CloudWatch
+# https://stackoverflow.com/a/45624044
```
Comment (jdalt) [inline src/titiler/application/titiler/application/main.py:52]: Was this issue caused by the upgrade or latent in our deployed titiler?

File: src/titiler/application/titiler/application/settings.py [hunk 4]
```diff
@@ -10,6 +10,7 @@ class ApiSettings(pydantic.BaseSettings):
     cors_origins: str = "*"
     cachecontrol: str = "public, max-age=3600"
     root_path: str = ""
+    path_prefix: str = ""
```
Comment (jdalt) [inline src/titiler/application/titiler/application/settings.py:13]: I don't love all the nested typed option stuff in titiler. I'm grateful you (or Jeff?) figured this out to make it work. It also looks like Pydantic might make classes in Python a bit nicer. It also seems like good Python is mostly typed.

File: pyproject.toml (hunk 2 above)
Comment (DanSchoppe) [reply pyproject.toml:34]: On SenteraLLC's fork, this worked similarly before:
https://github.com/SenteraLLC/titiler/blob/4accb097b8dfe0f9df5201d1e95b0e230857b77c/setup.py#L13

Unfortunately, with the upstream project's switch from `setup.py` -> `pyproject.toml`, I lost the ability to use a dynamic relative path like we had been before. Hence the code comment suggesting you update for your local filesystem if you want to run this locally.

File: src/titiler/application/titiler/application/main.py (hunk 3 above)
Comment (DanSchoppe) [reply src/titiler/application/titiler/application/main.py:52]: This is not a new contribution. It's actually a [common contribution](https://github.com/SenteraLLC/py-feature-set/blob/46e632af4f0e091d2c9262862b9ffa2588851df4/handlers.py#L17-L20) to Sentera's Python projects, as we've historically had trouble getting the Python Lambda logger to work with CloudWatch, probably due to the details in that StackOverflow link.

I can try getting rid of this if you'd like.

File: src/titiler/application/titiler/application/settings.py (hunk 4 above)
Comment (DanSchoppe) [reply src/titiler/application/titiler/application/settings.py:13]: Yeah, the stack is certainly not simple. I'm hoping this contribution in particular can die someday in favor of [a Mangum setting](https://github.com/developmentseed/titiler/pull/515#issuecomment-1332268220). That should reduce the burden of updating this fork quite a bit smaller.

> It also looks like Pydantic might make classes in Python a bit nicer. It also seems like good Python is mostly typed.

I'm also excited to see all the typing in this ecosystem. I've largely [cargo culted](https://en.wikipedia.org/wiki/Cargo_cult_programming) changes to this point, so I can't say I even understand it all that well.

File: src/titiler/application/titiler/application/main.py (hunk 3 above)
Comment (jdalt) [reply src/titiler/application/titiler/application/main.py:52]: This doesn't matter to me a lot. If you want to take a swing at it, I'm happy to review the solution.

**Review by jdalt — APPROVED** [review]

File: src/titiler/application/titiler/application/main.py (hunk 3 above)
Comment (DanSchoppe) [reply src/titiler/application/titiler/application/main.py:52]: Ok, I'll leave this for a python project that has better CI/CD and isn't such a pain to build/deploy/test. It's worth pushing on this at some point to make sure it's still necessary.

**Comment by nickb27** [comment]
QA 👍
